### PR TITLE
fix: webpack and esbuild query string support

### LINKED ISSUE #142: Query string is not included with id in transformInclude when using webpack
https://github.com/unjs/unplugin/blob/c1b73196e584e26d7fda497aad98d306b141bf43/src/webpack/index.ts#L51-L53

`id` here does not include query string.

It is possible to get query string when writing `use` as a function.

### PR DESCRIPTION
**Pull request by Demivan**
Fixes #142

### CODE REVIEW

**Comment by Demivan** [comment]
Would be nice if someone can verify this. Locally I get issue with some modules being transformed twice (vue custom blocks).

File: test/fixtures/transform/unplugin.js [hunk 1]
```diff
@@ -14,7 +23,14 @@ module.exports = createUnplugin((options) => {
         return null
       }
 
-      s.overwrite(index, index + '__UNPLUGIN__'.length, `[Injected ${options.msg}]`)
+      const injectedCode = `[Injected ${options.msg}]`
+
+      if (id.includes(injectedCode)) {
+        throw new Error('File was already transformed')
+      }
```
Comment (antfu) [inline test/fixtures/transform/unplugin.js:30]: I guess we might better commenting this out in production.

**Review by antfu — APPROVED** [review]

**Comment by antfu** [comment]
Using `use` as a function breaks Webpack 4 

- https://github.com/webpack/webpack/issues/8952
- https://github.com/webpack/webpack/issues/8406
- https://github.com/nuxt/bridge/issues/460

**Comment by Demivan** [comment]
Tried hacking using nuxt/bridge repo.

Discovered that ident will not work. Webpack 4 only uses ident if `use` is an array, not a function.
But using inline options as suggested in https://github.com/webpack/webpack/issues/8406 does not work either - loader is not being called for some reason. Investigating why.

**Comment by Demivan** [comment]
@antfu Found solution. PR incoming

**Pull request merged**